### PR TITLE
U2F token enrollment

### PR DESCRIPTION
**Pull request by Mipronimo**
The module privacyIDEA for simpleSAMLphp is now able to enroll u2f tokens in the authproc filter.
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/privacyidea/simplesamlphp-module-privacyidea/pull/53%23discussion_r249521443%22%2C%20%22https%3A//github.com/privacyidea/simplesamlphp-module-privacyidea/pull/53%23discussion_r252238485%22%2C%20%22https%3A//github.com/privacyidea/simplesamlphp-module-privacyidea/pull/53%23discussion_r252267126%22%2C%20%22https%3A//github.com/privacyidea/simplesamlphp-module-privacyidea/commit/d577235211f6dac660170d7650e76e294777441b%23commitcomment-32118225%22%2C%20%22https%3A//github.com/privacyidea/simplesamlphp-module-privacyidea/commit/d577235211f6dac660170d7650e76e294777441b%23commitcomment-32118235%22%2C%20%22https%3A//github.com/privacyidea/simplesamlphp-module-privacyidea/pull/53%23issuecomment-458975032%22%2C%20%22https%3A//github.com/privacyidea/simplesamlphp-module-privacyidea/pull/53%23issuecomment-458983912%22%5D%2C%20%22comments%22%3A%20%7B%22Pull%20679e238e4cdc4976721560a822b25623b9d7f57a%20templates/loginform.php%2019%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/privacyidea/simplesamlphp-module-privacyidea/pull/53%23discussion_r249521443%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Is%20there%20another%20way%20than%20the%20availability%20of%20the%20serial%20number%20to%20determin%20if%20this%20is%20the%20second%20step%20of%20the%20registration%20process%3F%22%2C%20%22created_at%22%3A%20%222019-01-21T16%3A50%3A16Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars1.githubusercontent.com/u/1908620%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/cornelinux%22%7D%7D%2C%20%7B%22body%22%3A%20%22Please%20have%20a%20look%20at%20this%20d577235%22%2C%20%22created_at%22%3A%20%222019-01-30T12%3A39%3A03Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars1.githubusercontent.com/u/35998229%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Mipronimo%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222019-01-30T14%3A01%3A28Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars1.githubusercontent.com/u/1908620%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/cornelinux%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20templates/loginform.php%3AL281-304%22%7D%2C%20%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/privacyidea/simplesamlphp-module-privacyidea/pull/53%23issuecomment-458975032%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%3E%20I%20would%20also%20like%20to%20take%20a%20look%20at%20the%20texts%20during%20registration.%5Cr%5Cn%5Cr%5Cn%40cornelinux%20it%20is%20done%20here%2047c07e1%22%2C%20%22created_at%22%3A%20%222019-01-30T15%3A00%3A01Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars1.githubusercontent.com/u/35998229%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Mipronimo%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%22%2C%20%22created_at%22%3A%20%222019-01-30T15%3A22%3A59Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars1.githubusercontent.com/u/1908620%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/cornelinux%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Commit%20d577235211f6dac660170d7650e76e294777441b%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/privacyidea/simplesamlphp-module-privacyidea/commit/d577235211f6dac660170d7650e76e294777441b%23commitcomment-32118225%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22lgtm%22%2C%20%22created_at%22%3A%20%222019-01-30T14%3A00%3A38Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars1.githubusercontent.com/u/1908620%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/cornelinux%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-complete%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222019-01-30T14%3A01%3A29Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars1.githubusercontent.com/u/1908620%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/cornelinux%22%7D%7D%5D%2C%20%22title%22%3A%20%22Commit%20d577235%20comments%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [x] <a href='#crh-comment-Pull 679e238e4cdc4976721560a822b25623b9d7f57a templates/loginform.php 19'></a> &nbsp;<b><a href='https://github.com/privacyidea/simplesamlphp-module-privacyidea/pull/53#discussion_r249521443'>File: templates/loginform.php:L281-304</a></b>
- <a href='https://github.com/cornelinux'></a> Is there another way than the availability of the serial number to determin if this is the second step of the registration process?
- <a href='https://github.com/Mipronimo'></a> Please have a look at this d577235
- [x] <a href='#crh-comment-Commit d577235211f6dac660170d7650e76e294777441b'></a> &nbsp;<b><a href='https://github.com/privacyidea/simplesamlphp-module-privacyidea/commit/d577235211f6dac660170d7650e76e294777441b#commitcomment-32118225'>Commit d577235 comments</a></b>
- <a href='https://github.com/cornelinux'></a> lgtm
- [x] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/privacyidea/simplesamlphp-module-privacyidea/pull/53#issuecomment-458975032'>General Comment</a></b>
- <a href='https://github.com/Mipronimo'></a> > I would also like to take a look at the texts during registration.
@cornelinux it is done here 47c07e1

<a href='https://www.codereviewhub.com/privacyidea/simplesamlphp-module-privacyidea/pull/53?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/privacyidea/simplesamlphp-module-privacyidea/pull/53?approve=1'></a>&nbsp;<a href='https://github.com/privacyidea/simplesamlphp-module-privacyidea/pull/53'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by cornelinux** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/privacyidea/simplesamlphp-module-privacyidea/pull/53'>Refresh</a>.<a href='#crh-update-none'></a>

File: templates/loginform.php [hunk 1]
```diff
@@ -280,13 +281,23 @@ class="float-l erroricon" style="margin: 15px "/>
...
     SimpleSAML_Logger::debug("signRequests json: " . json_encode($signRequests, TRUE));
     echo '<script type="text/javascript">';
-	echo 'var signRequests = ' . json_encode($signRequests) . ';';
-	echo '    u2f.sign(signRequests, function (result) {
-                console.log(result);
-                document.getElementById("signatureData").value = result.signatureData;
-                document.getElementById("clientData").value = result.clientData;
-                document.forms["piLoginForm"].submit();
-        });';
+    if (isset($this->data['serial'])) {
```
Comment (cornelinux) [inline templates/loginform.php]: Is there another way than the availability of the serial number to determin if this is the second step of the registration process?

Comment (Mipronimo) [reply templates/loginform.php]: Please have a look at this d577235

Comment (cornelinux) [inline templates/loginform.php]: :+1:<a href='#crh-update-none'></a>

**Comment by Mipronimo** [comment]
> I would also like to take a look at the texts during registration.

@cornelinux it is done here 47c07e1

**Comment by cornelinux** [comment]
:+1:

**Pull request merged**